src/internal/tester/its/m365.go

```diff
@@ -0,0 +1,209 @@
+package its
+
+import (
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/h2non/gock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/alcionai/corso/src/internal/common/idname"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/internal/tester/tconfig"
+	"github.com/alcionai/corso/src/pkg/account"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/count"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+	"github.com/alcionai/corso/src/pkg/services/m365/api/graph"
+)
+
+// ---------------------------------------------------------------------------
+// Gockable client
+// ---------------------------------------------------------------------------
+
+// GockClient produces a new exchange api client that can be
+// mocked using gock.
+func gockClient(creds account.M365Config, counter *count.Bus) (api.Client, error) {
+	s, err := graph.NewGockService(creds, counter)
+	if err != nil {
+		return api.Client{}, err
+	}
+
+	li, err := graph.NewGockService(creds, counter, graph.NoTimeout())
+	if err != nil {
+		return api.Client{}, err
+	}
+
+	return api.Client{
+		Credentials: creds,
+		Stable:      s,
+		LargeItem:   li,
+	}, nil
+}
+
+// ---------------------------------------------------------------------------
+// Intercepting calls with Gock
+// ---------------------------------------------------------------------------
+
+const graphAPIHostURL = "https://graph.microsoft.com"
+
+func V1APIURLPath(parts ...string) string {
+	return strings.Join(append([]string{"/v1.0"}, parts...), "/")
+}
+
+func InterceptV1Path(pathParts ...string) *gock.Request {
+	return gock.New(graphAPIHostURL).Get(V1APIURLPath(pathParts...))
+}
+
+// ---------------------------------------------------------------------------
+// Suite Setup
+// ---------------------------------------------------------------------------
+
+type IDs struct {
+	Provider          idname.Provider
+	ID                string
+	Email             string
+	DriveID           string
+	DriveRootFolderID string
+	TestContainerID   string
+	WebURL            string
+}
+
+// m365IntgTestSetup provides all the common references used in an m365 integration
+// test suite.  Call `its.GetM365()` to get the singleton for your test suite.
+// If you're looking for unit testing setup, use `uts.GetM365()` instead.
+type m365IntgTestSetup struct {
+	Acct     account.Account
+	Creds    account.M365Config
+	TenantID string
+
+	AC     api.Client
+	GockAC api.Client
```

Comment:
I think now that it's possible to mock out requesting credentials we should only have either a mocked client or a regular client. Having both makes it seem like it will make things error prone

Reply:
On one hand, the option for either one goes against the nature of the singleton here, where the (arbitrary) initial caller is expected to produce a live connection and fulfill all the queries.  On the other hand, I think you're right that we shouldn't need both.  The Gock client stuff should probably move out into its own package.  Also, we should probably treat gock'd tests as `unit` instead of `integration`, which would make it an irrelevant addition to this setup.

I'm happy to make these changes, but in the spirit of the scope of this PR I'm currently moving around architecture that's already in place.  Let me punt it to the end of the PR chain (so that I don't have to mess with conflicts) and I'll get these refactored.  Sound good?